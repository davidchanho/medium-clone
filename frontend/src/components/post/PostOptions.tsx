import React from "react";
import { Dropdown } from "react-bootstrap";
import { BsBookmark, BsThreeDots } from "react-icons/bs";
import { PostProps } from "../../types";
import { usePost } from "./usePost";

function PostOptions({ post }: PostProps) {
  const { onDeletePost } = usePost(post);

  return (
    <Dropdown className="mt-auto d-flex align-items-center justify-content-between">
      <Dropdown.ItemText className="p-0">
        <BsBookmark />
      </Dropdown.ItemText>

      <Dropdown.Toggle className="border-0 bg-white text-secondary shadow-none">
        <BsThreeDots />
      </Dropdown.Toggle>
      <Dropdown.Menu className="d-flex flex-column">
        <Dropdown.Item onClick={onDeletePost}>Dismiss This Post</Dropdown.Item>
      </Dropdown.Menu>
    </Dropdown>
  );
}

export default PostOptions;
