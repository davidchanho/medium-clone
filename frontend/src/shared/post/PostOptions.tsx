import React from "react";
import { Dropdown } from "react-bootstrap";
import { BsThreeDots } from "react-icons/bs";
import { PostProps } from "../../store/posts/types";
import Bookmark from "../bookmark";
import { usePost } from "./usePost";

function PostOptions({ post }: PostProps) {
  const { onDeletePost } = usePost(post);

  return (
    <Dropdown className="mt-auto d-flex align-items-center justify-content-between">
      <Dropdown.ItemText className="p-0">
        <Bookmark post={post} />
      </Dropdown.ItemText>

      <Dropdown.Toggle className="ml-2 bg-white text-secondary shadow-none">
        <BsThreeDots />
      </Dropdown.Toggle>
      <Dropdown.Menu className="d-flex flex-column">
        <Dropdown.Item onClick={onDeletePost}>Dismiss this story</Dropdown.Item>
        <Dropdown.Item onClick={onDeletePost}>Mute this author</Dropdown.Item>
        <Dropdown.Item onClick={onDeletePost}>
          Mute this publication
        </Dropdown.Item>
        <Dropdown.Item onClick={onDeletePost}>Report this story</Dropdown.Item>
        <Dropdown.Item onClick={onDeletePost}>Block this author</Dropdown.Item>
      </Dropdown.Menu>
    </Dropdown>
  );
}

export default PostOptions;
