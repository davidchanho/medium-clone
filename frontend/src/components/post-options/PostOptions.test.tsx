import React from "react";
import renderer from "react-test-renderer";
import PostOptions from ".";
import { testPost } from "../../tests";

describe("PostOptions", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PostOptions post={testPost} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
