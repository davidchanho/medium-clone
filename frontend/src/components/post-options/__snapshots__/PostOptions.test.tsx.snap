// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostOptions renders correctly 1`] = `
<div
  className="mt-auto d-flex align-items-center justify-content-between dropdown"
>
  <span
    className="p-0 dropdown-item-text"
  >
    <span
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="22"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M8 12l5 3V3a2 2 0 00-2-2H5a2 2 0 00-2 2v12l5-3zm-4 1.234l4-2.4 4 2.4V3a1 1 0 00-1-1H5a1 1 0 00-1 1v10.234z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </span>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="ml-2 bg-white text-secondary shadow-none dropdown-toggle btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M3 9.5a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm5 0a1.5 1.5 0 110-3 1.5 1.5 0 010 3z"
        fillRule="evenodd"
      />
    </svg>
  </button>
</div>
`;
